### DISCUSSION TITLE
Few questions I had about mini.hues.

### DISCUSSION THREAD

**Discussion by 231tr0n** [nvim-mini/mini.nvim#1253] [Q&A]
Hi echasnovski! I had these few questions in mind about mini.hues.
https://github.com/echasnovski/mini.nvim/issues/1156
The questions mostly are related to this issue.
1. Currently mini.hues uses bold as a differentiation for keywords. I wanted to know is there any way to generate a single color programatically for mini.hues using the Oklch algorithm and apply it to all possible places which affect source code being shown in bold?
2. I also was curious to know why was bold given a priority over a color for differentiation of keywords in code?
3. Can you add a new configuration option if you think it is fine to color those highlights instead of setting them bold?

I wanted to ask these questions since I found myself hard to differentiate keywords from normal text effectively. I tried all possible fonts and finally ended up with roboto which I think does the best job of doing the differentiation between bold and normal text and it is nowhere near close. I also tried mini.base16 but since it has less treesitter highlight groups as mentioned by you in https://github.com/echasnovski/mini.nvim/issues/1188, I had to settle with mini.hues.  I tried asking few other colleagues of mine as well just to make sure of it and their response was also the same and since I taught this could be a common concern I raised this discussion hoping to see if you could provide some solution about this issue.
![image](https://github.com/user-attachments/assets/1dfeb951-9e11-439c-89f6-dee51354aa82)

**Comment by echasnovski** [comment]
Hi, no worries! Thanks for creating a discussion instead of an issue.

> 1. Currently mini.hues uses bold as a differentiation for keywords. I wanted to know is there any way to generate a single color programatically for mini.hues using the Oklch algorithm and apply it to all possible places which affect source code being shown in bold?

Yes, it is possible. The general approach is the same as described in [this comment](https://github.com/echasnovski/mini.nvim/issues/1156#issuecomment-2295351010). Here is a rough outline:
- Set the color scheme based on 'mini.hues'.
- Retrieve the palette used in current color scheme via `local palette = MiniHues.make_palette(MiniHues.config)`. It will contain colors as [described here](https://github.com/echasnovski/mini.nvim/blob/848c5e8f428faf843051768e0d56104cd02aea1f/doc/mini-hues.txt#L294-L304).
- Tweak any highlight groups which you don't like. For bold keywords, use something like `vim.api.nvim_set_hl(0, 'Keyword', { fg = palette.orange })`. Use `:Inspect` to get the name of syntax highlight group under cursor. Another approach is to search ['lua/mini/hues.lua'](https://github.com/echasnovski/mini.nvim/blob/main/lua/mini/hues.lua) file for word "bold".

In theory, you can, of course, generate your own color based on the foreground and/or background and/or palette, but details are bound only by imagination. If it is what you want to do, I can recommend [`MiniColors.convert()`](https://github.com/echasnovski/mini.nvim/blob/848c5e8f428faf843051768e0d56104cd02aea1f/doc/mini-colors.txt#L785) for performing conversion.

> 2. I also was curious to know why was bold given a priority over a color for differentiation of keywords in code?

For several reasons:
- There are already quite a few colors used. Adding one extra means either making distinguishing less easy or not using extra colors for other stuff. And since I'd like for a limited set of distinctive foreground colors to be used in the whole color scheme, having bold allows it to be "an extra color", so to speak.
- Using bold serves as an ultimate fallback for people with color vision problems, since it will be distinguishable even in a totally greyscaled variant.

> 3. Can you add a new configuration option if you think it is fine to color those highlights instead of setting them bold?

Sorry, no. Using bold for keywords is one fundamental approach of color scheme which I think is very important. As there are ways to tweak that, I think this is acceptable.

**Reply by 231tr0n** [reply]
Yup thank you soo much!
